magellan-library/src/main/java/com/wealthfront/magellan/navigation/NavigationDelegate.kt

```diff
@@ -63,6 +69,16 @@ public open class NavigationDelegate(
     direction: Direction,
     backStackOperation: (Deque<NavigationEvent>) -> MagellanTransition
   ) {
+    navigationRequestHandler?.let { navRequestHandler ->
+      val backstackCopy = ArrayDeque(backStack)
+      backStackOperation.invoke(backstackCopy)
```

Comment:
this will do a "dry run" of the navigation operation on a copy of the backstack, from which it obtains a candidate currentNavigable